content/blog/2025/02/connecting-edi-to-the-enterprise-with-camel-and-smooks/index.md

```diff
@@ -0,0 +1,452 @@
+           <dataElement>00</dataElement>
+           <dataElement></dataElement>
+           <dataElement>ZZ</dataElement>
+           <dataElement>ACME</dataElement>
+           <dataElement>ZZ</dataElement>
+           <dataElement>MYCORP</dataElement>
+           <dataElement>071216</dataElement>
+           <dataElement>1406</dataElement>
+           <dataElement>U</dataElement>
+           <dataElement>00204</dataElement>
+           <dataElement>000000263</dataElement>
+           <dataElement>1</dataElement>
+           <dataElement>T</dataElement>
+           <dataElement>&gt;</dataElement>
+       </segment>
+       ...
+       <segment>
+           <segmentId>IEA</segmentId>
+           <dataElement>1</dataElement>
+           <dataElement>000000263</dataElement>
+       </segment>
+   </interchange>
+   ```
+
+   Each emitted **event** corresponds to a segment or data element within the EDI document. Within `ingest-x12-config.xml`, Smooks is configured to **bind** these events to POJOs using **visitors** which target the events with **selectors**. Different visitors then **route** the POJOs via Camel to (a) verify the document identifiers with a Trading Partner Management system, AND (b) send them to an ERP for purchase order fulfillment.
+
+4. The Smooks endpoint produces an XML representation of the EDI purchase order, even if a non-fatal error happens while binding the events or routing the beans. The _Ingest Purchase Order_ route goes on to post this XML to a tracking system for transaction visibility and monitoring purposes.
+
+5. To complete the circle, the _Generate Functional Ack_ route transmits an X12 acknowledgement of the transaction over AS2 to the customer. In particular, the route takes the POJOs, which were derserialised from EDI segments in the previous Smooks execution, and sends them to another Smooks endpoint. This endpoint calls Smooks with the `gen-x12-ack-config.xml` config to:
+    1. Generate the acknowledgment event stream by materialising an XML template with [FreeMarker](https://freemarker.apache.org/) (a popular template engine),
+    2. Record the acknowledgement via Camel, and
+    3. Serialise the acknowledgement to EDI
+
+---
+
+We will now dive into the individual YAML routes and Smooks XML configs orchestrating the B2B exchange.
+
+#### Receive Purchase Order Route
+
+The following route receives the purchase order from the customer:
+
+```yaml
+- routeConfiguration:
+    id: as2Error
+    onException:
+      - onException:
+          handled:
+            constant: true
+          exception:
+            - java.lang.Exception
+          steps:
+            - script:
+                groovy: |
+                  httpContext = exchangeProperties['CamelAs2.interchange']
+                  httpContext.setAttribute(org.apache.camel.component.as2.api.AS2Header.DISPOSITION_TYPE,
+                                           org.apache.camel.component.as2.api.entity.AS2DispositionType.FAILED)
```

Comment:
Seems that the code syntax highlighting inserts the trailing whitespace, this line is rendered as (`$` to mark EOL):

```
<span class="sd">                                           org.apache.camel.component.as2.api.entity.AS2DispositionType.FAILED)</span><span class="w">                  $
```

Not sure how to configure that, it could be an issue within the syntax highlighting that we can't address at the moment.
Perhaps the best thing for now is to add `public/blog/2025/02/connecting-edi-to-the-enterprise-with-camel-and-smooks/index.html` to `.htmlvalidateignore`.

Reply:
So we create this file in the root folder (incl the starting dot) of camel-website, and then add that link ?

Reply:
Should be able to add:
```
public/blog/2025/02/connecting-edi-to-the-enterprise-with-camel-and-smooks/index.html
```
to the [existing file](https://github.com/apache/camel-website/blob/3a60d0142c49b731664dcf3df57a4a924ed0c0f0/.htmlvalidateignore)

Reply:
doh somehow I did not grep that file, thanks